docusaurus/docs/reactnative/guides/live-location-sharing.mdx

```diff
@@ -0,0 +1,635 @@
+---
+id: live-location-sharing
+title: Live location Sharing Message
+---
+
+In this cookbook, we will build a simple live location sharing message feature using [custom attachments](./custom-attachment.mdx).
+
+The goal is to create two screens:
+
+* Chat screen with the live location shown on map in a message
+* A detailed map screen that is shown when the live location is tapped
+
+## Setup
+
+### Install Dependencies
+
+Run the following commands in your terminal of choice:
+
+```bash
+yarn add @react-native-community/geolocation
+yarn add react-native-maps
+```
+
+* `@react-native-community/geolocation` library is used to watch the current location of the user and then send it in the message
+* `react-native-maps` is used to display the location of the user using the native maps present on iOS and Android 
+
+### Configure location permissions
+
+**iOS**
+
+You need to include `NSLocationWhenInUseUsageDescription` and `NSLocationAlwaysAndWhenInUseUsageDescription` in `Info.plist` to enable geolocation when using the app.
+
+**Android**
+
+To request access to location, you need to add the following line to your app's `AndroidManifest.xml`:
+
+```xml
+<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+```
+
+### Android: Add API Key
+
+On Android, one has to use [Google Maps](https://developers.google.com/maps/documentation/), which in turn requires you to obtain an [API key for the Android SDK](https://developers.google.com/maps/documentation/android-sdk/signup). On iOS, the native Apple Maps implementation is used and API keys are not necessary.
+
+Add your API key to your manifest file (`android/app/src/main/AndroidManifest.xml`):
+
+```xml
+<application>
+   <!-- You will only need to add this meta-data tag, but make sure it's a child of application -->
+   <meta-data
+     android:name="com.google.android.geo.API_KEY"
+     android:value="Your Google maps API Key Here"/>
+</application>
+```
+
```

Comment:
Let's add an intro to how our solution works.

```suggestion
## Implementation outline

We will use a custom attachment with a `type` set to `location`. This custom attachment will be part of the message the app sends to the channel. This attachment will carry the latest longitude, latitude, and timestamp when the location sharing has been stopped.

On the sharer side, the application itself will poll for location updates and whenever a new location is emitted, it will update the appropriate message/attachment through our `channel.updateMessage()` APIs.

On the recipient side, the application will listen for `message.updated` events and will update the interactive maps accordingly.
```